--- riak.schema.origin	2024-06-28 12:50:20
+++ riak.schema	2024-06-28 12:50:20
@@ -11,7 +11,14 @@
 %% and atoms. The atoms denote the keys for retrieving metadata from the logger events.
 %% More information on default metadata can be found here (https://www.erlang.org/doc/man/logger_formatter.html#type-template).
 {mapping, "logger.format", "kernel.logger", [
-  {default, "[time,\" [\",level,\"] \",pid,\"@\",mfa,\":\",line,\" \",msg,\"\\n\"]."}
+  {default, [
+    "{\"time\": \"", time, "\", ",
+    "\"level\": \"", level, "\", ",
+    "\"pid\": \"", pid, "\", ",
+    "\"mfa\": \"", mfa, "\", ",
+    "\"line\": ", line, ", ",
+    "\"msg\": \"", msg, "\"}\n"
+  ]}
 ]}.

 %% @doc Filename to use for log files.
